File: Day290_2025/Q290_GFG.java
class Node {
    int data;
    Node left;
    Node right;

    Node(int data) {
        this.data = data;
    }
}

class Q290_GFG {
    private static int solve(Node root, int sum) {
        if (root == null) return 0;

        int right = solve(root.right, sum);
        int left = solve(root.left, root.data + right + sum);

        int total = root.data + left + right;
        root.data = right + sum;

        return total;
    }

    public static void transformTree(Node root) {
        solve(root, 0);
    }
}

File: Day290_2025/Q290_GFG.cpp
/*
class Node {
  public:
    int data;
    Node* left;
    Node* right;

    Node(int value) {
        data = value;
        left = nullptr;
        right = nullptr;
    }
};*/

class Solution {
    int solve(Node* root, int sum){
        if(!root)return 0;
        int r = solve(root->right, sum);
        int l = solve(root->left, root->data+r+sum);
        int store = root->data+l+r;
        root->data = r+sum;
        return store;
    }
  public:
    void transformTree(Node *root) {
        // Q. BST to greater sum tree
        // code here
        solve(root,0);
    }
};